src/roslibpy/comm/comm_cli.py

```diff
@@ -328,6 +331,65 @@ def call_in_thread(self, callback):
         """
         Task.Factory.StartNew(callback, self.cancellation_token)
 
+    def blocking_call_from_thread(self, callback, timeout):
+        """Call the given function from a thread, and wait for the result synchronously
+        for as long as the timeout will allow.
+
+        Args:
+            callback: Callable function to be invoked from the thread.
+            timeout (:obj: int): Number of seconds to wait for the response before
+                raising an exception.
+
+        Returns:
+            The results from the callback, or a timeout exception.
+        """
+        auto_event = AutoResetEvent(False)
+        result_placeholder = {'auto_event': auto_event}
+        ThreadPool.QueueUserWorkItem(WaitCallback(callback), result_placeholder)
+        if timeout and auto_event.WaitOne(timeout * 1000) or not timeout and auto_event.WaitOne():
```

Comment:
The `.WaitOne()` calls, (or `.Wait()` if you change to `ManualResetEventEventSlim`) should take the `self.cancellation_token` as parameter, which handles cancellation of all waiting threads if triggered (see `def terminate()`).

Reply:
Indeed it should.  Thanks.